pennylane/templates/subroutines/qft.py

```diff
@@ -70,17 +70,19 @@ def circuit_qft(basis_state):
     def num_params(self):
         return 0
 
-    @property
-    def matrix(self):
-        # Redefine the property here to allow for a custom _matrix signature
-        mat = self._matrix(len(self.wires))
+    def matrix(self, wire_order=None):
+        canonical_matrix = self.compute_matrix(len(self.wires))
         if self.inverse:
-            mat = mat.conj()
-        return mat
+            canonical_matrix = canonical_matrix.conj()
+
+        if wire_order is None or self.wires == qml.wires.Wires(wire_order):
+            return canonical_matrix
 
-    @classmethod
+        return canonical_matrix
+
+    @staticmethod
     @functools.lru_cache()
-    def _matrix(cls, num_wires):
+    def compute_matrix(num_wires):
         dimension = 2 ** num_wires
```

Comment:
Another reason to allow for custom signatures: some params and hyperparams are unhashable, in which case we may be happy about the flexibility